test/XRP/xrp-client-integration.test.ts

```diff
@@ -229,4 +229,71 @@ describe('XrpClient Integration Tests', function (): void {
       AccountRootFlags.checkFlag(AccountRootFlags.LSF_DEPOSIT_AUTH, flags!),
     )
   })
+
+  it('Authorize Sending Account - failure on authorizing self', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account
+    // WHEN authorizeSendingAccount is called with the account's own address
+    const result = await xrpClient.authorizeSendingAccount(
+      wallet.getAddress(),
+      wallet,
+    )
+
+    // THEN the transaction fails due to a malformed transaction.
+    const transactionHash = result.hash
+    const transactionStatus = result.status
+
+    assert.exists(transactionHash)
+    assert.equal(transactionStatus, TransactionStatus.MalformedTransaction)
+  })
+
+  it('Authorize Sending Account - failure on authorizing already authorized account', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account that has authorized another account to send funds to it
+    const sendingWallet = await XRPTestUtils.randomWalletFromFaucet()
+    await xrpClient.enableDepositAuth(wallet)
+    await xrpClient.authorizeSendingAccount(sendingWallet.getAddress(), wallet)
+
+    // WHEN authorizeSedingAccount is called on the already authorized account
+    const result = await xrpClient.authorizeSendingAccount(
+      sendingWallet.getAddress(),
+      wallet,
+    )
+
+    // THEN the transaction fails and the cost of the transaction is claimed by the network.
+    const transactionHash = result.hash
+    const transactionStatus = result.status
+
+    assert.exists(transactionHash)
+    assert.equal(transactionStatus, TransactionStatus.ClaimedCostOnly)
+  })
+
+  it('Authorize Sending Account - can send funds after authorize', async function (): Promise<
```

Comment:
I'm wondering if we can add to this test a check where the account in question first fails to be able to send?  So like: 
deposit authorization is enabled
our sending account tries to send money and things are rejected
we authorize the account
then it succeeds

Reply:
yeah, i was thinking about this - i actually want to add that as an explicit test for enableDepositAuth. lmk what you think